base/test.jl

```diff
@@ -908,4 +909,16 @@ function detect_ambiguities(mods...; imported::Bool=false)
     collect(ambs)
 end
 
+"""
+The `GenericString` can be used to test generic string APIs that program to
+the `AbstractString` interface. In order to ensure that functions can work
+with string types besides the standard `String` type.
+"""
+immutable GenericString <: AbstractString
```

Comment:
But my point is that *this* file is full of similar utilities: `@test_approx_eq`, `detect_ambiguities`, `@inferred`, etc.